app/uk/gov/nationalarchives/omega/editorial/services/ReferenceDataService.scala

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright (c) 2022 The National Archives
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package uk.gov.nationalarchives.omega.editorial.services
+
+import com.google.inject.ImplementedBy
+import uk.gov.nationalarchives.omega.editorial.models._
+
+@ImplementedBy(classOf[StubReferenceDataService])
+trait ReferenceDataService {
```

Comment:
Do we need this trait with a default implementation?

I'm aware that we override this trait in the tests, however I think we could equally make test reference data class extend `StubReferenceDataService` - as long we don't do side effecting work in the constructor (which I would flag on review anyway).

I think that creating a trait where the only non-default implementation is in the tests seems a bit like altering the application code for the test's sake. It's also a pain to have to change the abstract method implementation every time a function signature changes.

Reply:
I'll look into it. But it's a pattern I've seen many times elsewhere.

Reply:
I tried it - you're right; the trait isn't needed.

Will change.

Reply:
Changed.